docs/components/renderer.rst

```diff
@@ -6,5 +6,51 @@
 Renderer
 ========
 
-..  warning::
-    This needs to be documented
+A Renderer transforms the :abbreviation:`AST (abstract syntax tree)` into a
+desired output format.
+
+The most common renderers handle each document separately. For example the
+:php:class:`\phpDocumentor\Guides\Renderer\HtmlRenderer` renders the AST into
+HTML.
+
+Each renderer must implement
+:php:interface:`\phpDocumentor\Guides\Renderer\TypeRenderer`. Basic document
+type renderers like the HTML or Latex renderer can extend
+the :php:class:`\phpDocumentor\Guides\Renderer\BaseTypeRenderer`.
+
+All renderers must be registered in the ContainerConfigurator of the extension
+with the tag :php:`'phpdoc.renderer.typerenderer'` and additional format information.
+
+Example: a plaintext renderer
+=============================
+
+Create a class called :php:`PlaintextRenderer` which extends
+:php:class:`\phpDocumentor\Guides\Renderer\BaseTypeRenderer` and is
+empty otherwise.
```

Comment:
Actually, I don't understand the "and is empty otherwise". Why otherwise? I wouldn't write "empty", maybe better "which just extends BaseTypeRenderer" and omit the "empty otherwise".

Reply:
good point!